ddl/db_integration_test.go

```diff
@@ -527,24 +527,43 @@ func (s *testIntegrationSuite) TestNullGeneratedColumn(c *C) {
 func (s *testIntegrationSuite) TestChangingCharsetToUtf8(c *C) {
 	tk := testkit.NewTestKit(c, s.store)
 
-	tk.MustExec("USE test")
+	tk.MustExec("use test")
+	tk.MustExec("create table t1(a char(10) charset utf8)")
+	tk.MustExec("alter table t1 modify column a char(10) charset utf8mb4")
+
 	tk.MustExec("create table t(a char(10) charset latin1)")
-	tk.MustExec("alter table t modify column a char(10) charset latin1")
-	tk.MustExec("alter table t modify column a char(10) charset utf8")
-	tk.MustExec("alter table t modify column a char(10) charset utf8mb4")
-	rs, err := tk.Exec("alter table t modify column a char(10) charset utf8mb4 collate utf8bin")
+	rs, err := tk.Exec("alter table t modify column a char(10) charset latin1")
+	if rs != nil {
+		rs.Close()
+	}
+	c.Assert(err, IsNil)
+	rs, err = tk.Exec("alter table t modify column a char(10) charset utf8")
 	if rs != nil {
 		rs.Close()
 	}
 	c.Assert(err, NotNil)
-	tk.MustExec("alter table t modify column a char(10) charset utf8mb4 collate utf8mb4_bin")
-	rs, err = tk.Exec("alter table t modify column a char(10) charset utf8 collate utf8_bin")
+
+	rs, err = tk.Exec("alter table t modify column a char(10) charset utf8mb4")
```

Comment:
How about use `testErrorCode` ?

Reply:
Oh, `errUnsupportedModifyCharset` testErrorCode is not mysql error code.